src/fides/api/ops/models/privacy_preference.py

```diff
@@ -0,0 +1,206 @@
+# pylint: disable=R0401, C0302
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, Type
+
+from sqlalchemy import ARRAY, Column
+from sqlalchemy import Enum as EnumColumn
+from sqlalchemy import ForeignKey, String, UniqueConstraint
+from sqlalchemy.dialects.postgresql import JSONB
+from sqlalchemy.ext.mutable import MutableDict, MutableList
+from sqlalchemy.orm import Session, relationship
+from sqlalchemy_utils import StringEncryptedType
+from sqlalchemy_utils.types.encrypted.encrypted_type import AesGcmEngine
+
+from fides.api.ops.common_exceptions import PrivacyNoticeHistoryNotFound
+from fides.api.ops.db.base_class import JSONTypeOverride
+from fides.api.ops.models.privacy_notice import (
+    PrivacyNotice,
+    PrivacyNoticeHistory,
+    PrivacyNoticeRegion,
+)
+from fides.api.ops.models.privacy_request import PrivacyRequest, ProvidedIdentity
+from fides.core.config import CONFIG
+from fides.lib.db.base import Base  # type: ignore[attr-defined]
+
+
+class UserConsentPreference(Enum):
+    opt_in = "opt_in"  # The user wants to opt in to the notice
+    opt_out = "opt_out"  # The user wants to opt out of the notice
+    acknowledge = "acknowledge"  # The user has acknowledged this notice
+
+
+class RequestOrigin(Enum):
+    privacy_center = "privacy_center"
+    privacy_modal = "privacy_modal"
+
+
+class PrivacyPreferenceHistory(Base):
+    """The DB ORM model for storing PrivacyPreferenceHistory, used for saving
+    every time consent preferences are saved for reporting purposes.
+
+    add fides_user_provided_identity nullable FK
+    add fides_user_device_id encrypted field
+    add ip address
+    add consent preference method?
+    """
```

Comment:
This new `PrivacyPreferenceHistory` record should be created every single time a preference is saved for consent reporting.  It also is the primary record used in consent propagation.    The main goal here is to be storing as much relevant information as possible here to reduce the number of joins for reporting.

There are other fields that I need to add to this record for reporting, but as I need to get clarity on some of them, I will further expand this table as part of that work.